.github/workflows/cypress-workflow-snapshot-based.yml

```diff
@@ -0,0 +1,86 @@
+name: Snapshot based E2E Cypress tests workflow
+on:
+  pull_request:
+    branches:
+      - main
+      - dev-*
+  push:
+    branches:
+      - main
+      - dev-*
+jobs:
+  tests:
+    name: Run Cypress E2E tests
+    runs-on: ubuntu-latest
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - name: Get and run OpenSearch
+        run: |
+          wget https://artifacts.opensearch.org/snapshots/core/opensearch/1.0.0-SNAPSHOT/opensearch-min-1.0.0-SNAPSHOT-linux-x64-latest.tar.gz
```

Comment:
nonblocking: create a variable for opensearch version "1.0.0" to make it easy for future releases. In this way, there is no need to modify the "1.0.0" in multiple places.

Reply:
That makes sense, I can make that change in a future PR!